tests/src/python/providertestbase.py

```diff
@@ -120,6 +120,51 @@ def testGetFeaturesCompiled(self):
         except AttributeError:
             print 'Provider does not support compiling'
 
+    def testOrderBy(self):
+        request = QgsFeatureRequest().addOrderBy('cnt')
+        values = [f['cnt'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [-200, 100, 200, 300, 400])
+
+        request = QgsFeatureRequest().addOrderBy('cnt', False)
+        values = [f['cnt'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [400, 300, 200, 100, -200])
+
+        request = QgsFeatureRequest().addOrderBy('name')
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, ['Apple', 'Honey', 'Orange', 'Pear', NULL])
+
+        request = QgsFeatureRequest().addOrderBy('name', True, True)
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [NULL, 'Apple', 'Honey', 'Orange', 'Pear'])
+
+        request = QgsFeatureRequest().addOrderBy('name', False)
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, [NULL, 'Pear', 'Orange', 'Honey', 'Apple'])
+
+        request = QgsFeatureRequest().addOrderBy('name', False, False)
+        values = [f['name'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, ['Pear', 'Orange', 'Honey', 'Apple', NULL])
+
+        # Case sensitivity
+        request = QgsFeatureRequest().addOrderBy('name2')
+        values = [f['name2'] for f in self.provider.getFeatures(request)]
+        self.assertEquals(values, ['Apple', 'Honey', 'NuLl', 'oranGe', 'PEaR'])
```

Comment:
Bah...!

My thoughts... Either:
- just make a note of this in the docs, something along the lines of "Note that behavior of case sensitivity while sorting depends on the platform, user's locale and provider support."
- add extra flags for "ignore case during sorting" , "lowercase before uppercase"

I'm wondering now if it's more future proof to make the ascending and null first boolean arguments to OrderByClause instead use flags rather than seperate arguments. Thoughts?